airflow/contrib/hooks/emr_hook.py

```diff
@@ -36,6 +36,26 @@ def get_conn(self):
         self.conn = self.get_client_type('emr', self.region_name)
         return self.conn
 
+    def get_cluster_id_by_name(self, emr_cluster_name):
+        conn = self.get_conn()
+
+        response = conn.list_clusters(
+            ClusterStates=[
+                'RUNNING', 'WAITING'
```

Comment:
I would suggest making the list of valid states to be a parameter -- in the past I have done something like this and included `'BOOTSTRAPPING'` in my list.